posts/2024-09-05_writing_my_first.../writing_my_first_custom_ci_cd_action_for_the_pharmaverseblog.qmd

```diff
@@ -0,0 +1,187 @@
+---
+title: "Writing my first custom CICD action for the pharmaverseblog"
+author:
+  - name: Edoardo Mancini
+description: "I recently wrote my first custom CICD pipeline for the pharmaverseblog, in an effort to police the use of post tags within the blog. These are my learnings and experiences!"
+date: "2024-09-05"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [Technical]
+# feel free to change the image
+image: "why-cicd.jpg"
+
+---
+
+<!--------------- typical setup ----------------->
+
+```{r setup, include=FALSE}
+long_slug <- "2024-09-05_writing_my_first..."
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here ----------------->
+
+# The problem 
+
+Each pharmaverseblog post is tagged with one or more categories that describe the topics discussed within it. For instance, this post is tagged **Technical**. When making a new blog post, users are invited to select the tags from a curated list designed to split the posts according to categories that balance specificity and generality. Here is the list we currently use:
+
+```{r, eval = FALSE}
+c("Metadata", "SDTM", "ADaM", "TLG", "Shiny", "Community", "Conferences", "Submissions", "Technical")
+```
+
+Users **can** add to this list, however we have observed that more often than not, if users do diverge from it is due to a typo. This has unfortunate effects within our pharmaverseblog, chiefly that our front page glossary of posts is now split:
+
+![](mismatched_categories.png){fig-align="center" width="140"}
+
+Within the pharmaverseblog editor team, we wondered whether there was a simple way to police these tags a bit more, perhaps in an automated manner. Enter CICD checks! 
+
+::: {.callout-tip}
+## What is CICD?
+
+CICD stands for __Continuous Integration, Continuous Deployment__ and is a catch-all term for automated code pipelines that ensure that new code added to an existing codebase seamlessly assimilates with the rest of the codebase without introducing unexpected behavior. 
+
+Often, when working in R projects hosted on GitHub, one encounters CICD in the form of checks that are triggered when making a pull request to the `main` branch (see image below for an example). These can check all sorts of aspects, ranging from correct spelling in your documentation all the way to executing your function unit tests and checking they all pass.
+
+![](cicd_pipeline.png){fig-align="center" width="600"}
+
+:::
+
+The pharmaverseblog already had three active CICD pipelines, which run for spelling, style and links. We did not write the code for these checks ourselves, rather we just activated open-source checks for our code repository. But what if we could write our own, **custom CICD pipeline** to check that new blog posts use tags from our selected list?
```

Comment:
The spelling CICD is a custom pipeline.

Reply:
thanks, updated

Reply:
The spelling pipeline was written by me :)

Reply:
credited you for it 😄

Reply:
Haha thank you! 
This is not what I meant though, it still showed the old version for me, where you said that there are no custom pipelines, I think I saw the file where you did not update this yet :O